infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/SessionManager.java

```diff
@@ -384,12 +383,10 @@ private void updateRouterWithNewSessions() {
             } else {
                 router.getSessionToRemoteClusterDescriptor()
                         .put(session, topology.getRemoteSinkClusters().get(session.getSinkClusterId()));
-                router.getSessionToOutstandingRequests().put(session, new HashMap<>());
             }
 
             if(router.isConnectionStarterForSession(session)) {
-                router.getSessionToLeaderConnectionFuture().put(session, new CompletableFuture<>());
-                router.getSessionToOutstandingRequests().putIfAbsent(session, new HashMap<>());
```

Comment:
no...we now have a SessionToOutstandingRequests.putIfAbsent() in the router.sendRequestAndGetCompletable() : https://github.com/CorfuDB/CorfuDB/pull/3664/files#diff-d7618452e9e64d5d1bd6b0074898de835dfdf156c16bf832af194288a9a45f86R324

This change was needed because we now have a leadership_loss msg sent in parallel to the stopReplication for  a session. On stopReplication() we clear the in-memory data as well, so SessionToOutstandingRequests can now be empty while the source tries to send a leadership_loss msg, throwing an NPE trying to get the next ID for the outgoing request.
This approach resolves this case